jujugui/static/gui/src/app/components/machine-view/machine/machine.js

```diff
@@ -88,30 +202,47 @@ YUI.add('machine-view-machine', function() {
       @returns {Object} the machine hardware elements.
     */
     _generateHardware: function(unitCount) {
-      if (this.props.type === 'container' || !this.props.showConstraints) {
+      if (this.props.type === 'container' || !this.props.showConstraints ||
+          this.state.showForm) {
         return;
       }
-      var machine = this.props.machine;
-      var hardware = machine.hardware;
-      var hardwareDetails;
-      if (hardware) {
-        var cpu = hardware.cpuPower;
-        var disk = hardware.disk;
-        var mem = hardware.mem;
-        var cpuCores = hardware.cpuCores;
-        if (cpuCores && cpu && disk && mem) {
+      const machine = this.props.machine;
+      const hardware = machine.hardware || this._getConstraints() || {};
+      let hardwareDetails;
+      let cpu = hardware.cpuPower;
+      let disk = hardware.disk;
+      let mem = hardware.mem;
+      let cpuCores = hardware.cpuCores;
+      if (cpuCores || cpu || disk || mem) {
+        let details = [];
+        if (cpuCores) {
```

Comment:
I wonder if these checks to generate this string shouldn't be moved to a util as you've duplicated it twice now.